src/reqwest/src/async/nodes_info.rs

```diff
@@ -0,0 +1,145 @@
+/*! Contains the `NodesInfoResponse` type for sniffing node addresses in the cluster. */
+
+use std::fmt;
+use serde::de::{Deserialize, Deserializer, Visitor, MapAccess};
+use res::parsing::{IsOk, HttpResponseHead, Unbuffered, MaybeOkResponse, ResponseBody};
+use res::error::ParseResponseError;
+
+use std::iter::IntoIterator;
+use std::vec::IntoIter;
+
+#[derive(Debug, PartialEq, Deserialize)]
+pub struct NodesInfoResponse {
+    #[serde(deserialize_with = "deserialize_nodes")]
+    nodes: Vec<SniffedNode>,
+}
+
+#[derive(Debug, PartialEq, Deserialize)]
+pub struct SniffedNode {
+    pub http: Option<SniffedNodeHttp>,
+}
+
+#[derive(Debug, PartialEq, Deserialize)]
+pub struct SniffedNodeHttp {
+    pub publish_address: Option<String>,
+}
+
+impl IntoIterator for NodesInfoResponse {
+    type Item = SniffedNode;
+    type IntoIter = IntoIter<SniffedNode>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.nodes.into_iter()
+    }
+}
+
+impl IsOk for NodesInfoResponse {
```

Comment:
It would be nice to have a `DefaultIsOk` marker trait that automatically implements `IsOk` with this body. It saves a bunch of `use` statements.